src/components/scenes/Loans/LoanCloseScene.tsx

```diff
@@ -40,32 +45,62 @@ export const LoanCloseScene = (props: Props) => {
   const { loanAccountId } = route.params
   const loanAccount = loanAccounts[loanAccountId]
   const { borrowPlugin, borrowEngine: initBorrowEngine } = loanAccount
+  const borrowPluginId = borrowPlugin.borrowInfo.borrowPluginId
 
   // Async State:
   // Refreshing borrowEngine TODO: refactor common method
   const borrowEngineRefresher = React.useCallback(async () => borrowPlugin.makeBorrowEngine(initBorrowEngine.currencyWallet), [borrowPlugin, initBorrowEngine])
   const borrowEngine = useRefresher<BorrowEngine>(borrowEngineRefresher, initBorrowEngine, 10000)
-  const [approvableAction, approvableActionError] = useAsyncValue(async () => borrowEngine.close(), [borrowEngine])
-
-  // Derived State:
-  const networkFee = approvableAction != null ? approvableAction.networkFee.nativeAmount : '0'
-  const { currencyWallet: wallet } = borrowEngine
+  const { currencyWallet: borrowEngineWallet } = borrowEngine
 
   const collaterals = useWatch(borrowEngine, 'collaterals').filter(collateral => !zeroString(collateral.nativeAmount))
+  const collateral = collaterals.find(collateral => !zeroString(collateral.nativeAmount))
+  const collateralTokenId = collateral?.tokenId
   const debts = useWatch(borrowEngine, 'debts').filter(debt => !zeroString(debt.nativeAmount))
+  const debt = debts.find(debt => !zeroString(debt.nativeAmount))
+  const debtTokenId = debt?.tokenId
+
   const isLoanCloseSupported = collaterals.length === 1 && debts.length === 1
 
-  // Handlers:
-  const onSliderComplete = async (reset: () => void) => {
-    if (approvableAction == null) return
+  // Derived State:
+  const [repayApprovalAction, repayApprovalActionError] = useAsyncValue<ApprovableAction | null>(async () => {
+    if (collateralTokenId == null || debtTokenId == null) return null
+    return await borrowEngine.repayWithCollateral({ debtTokenId, collateralTokenId })
+  }, [borrowEngine, debtTokenId, collateralTokenId])
+  const [withdrawApprovalAction, withdrawApprovalActionError] = useAsyncValue<ApprovableAction | null>(async () => {
+    if (debtTokenId == null) return null
+    return await borrowEngine.withdraw({ tokenId: collateralTokenId, nativeAmount: collateral?.nativeAmount })
+  }, [borrowEngine, debtTokenId])
+  const isApprovableActionValid = repayApprovalAction != null && withdrawApprovalAction != null
 
-    if (checkLoanHasFunds(borrowEngine)) {
```

Comment:
Why was the check removed?

Reply:
There was a bug where if the network was down (frequent with MATIC), the borrow engine would see 0 debts/collaterals, and close the loan account without doing anything.

If the slider was enabled, the request should be valid, according to what the scene knows about the state of the borrowEngine. 

If something changed that makes the approval fail that happens between when the user started sliding an enabled slider and when the slider completes, that ought to just show up as an error. This last point is probably worth adding as a change.